src/components/ProductDescription/QuantityTextField.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from "react";
+
+import { TextField } from "@components/molecules";
+
+interface QuantityTextFieldProps {
+  quantity: number;
+  maxQuantity: number;
+  onQuantityChange: (value: number) => void;
+  hideErrors: boolean;
+}
+
+export const QuantityTextField: React.FC<QuantityTextFieldProps> = ({
+  quantity,
+  maxQuantity,
+  onQuantityChange,
+  hideErrors,
+}: QuantityTextFieldProps) => {
+  const [tempQuantity, setTempQuantity] = useState<number | string>(quantity);
+  const [isTooMuch, setIsTooMuch] = useState(false);
+
+  const handleBlurQuantityInput = () => {
+    const newQuantity =
+      typeof tempQuantity === "number"
+        ? tempQuantity
+        : parseInt(tempQuantity, 10);
```

Comment:
Parseint returns NaN for NaNs, dont think there's need for checking that again.

Reply:
I'm not sure what you mean. `tempQuantity === "number"` is needed, because tempQuantity might be `number` or `string`, while the case is that `parseInt()` accepts just string as argument.

Reply:
Fixed